client/components/tinymce/plugins/wpcom-view/views/embed/index.js

```diff
@@ -46,11 +41,11 @@ export default class EmbedViewManager extends EventEmitter {
 		}
 	}
 
-	addListener( event, listener ) {
+	addListener( event, listener, store ) {
 		super.addListener( event, listener );
-
 		if ( 'change' === event && 1 === this.listeners( event ).length ) {
-			sites.addListener( event, this.sitesListener );
+			this.store = store;
+			this.unsubscribeStore = store.subscribe( this.sitesListener );
```

Comment:
What we want isn't too dissimilar to the hack we added to get around `sites.once` with the `SITES_ONCE_CHANGED` event. The tricky part here is for the middleware to have access to the siteListener, it might be reasonable to dispatch a new event like:

Perhaps:
```jsx
dispatch( {
		type: SITES_ANY_CHANGED,
		listener: selectOnSitesChange,
} );
```

The middleware would then need to be updated to have a second object of listeners that we can manage. Right now we clear them out each time for any once handlers https://github.com/Automattic/wp-calypso/blob/master/client/state/lib/middleware.js#L150

If I'm not super clear in explaining this, maybe @mcsf can take a shot.